src/messages.h.jinja

```diff
@@ -0,0 +1,42 @@
+// NOTE: This file has been generated automatically by “update-message-registry.py”.
+//       Do not edit manually!
+
+#ifndef _XKBCOMMON_MESSAGE_H_
+#define _XKBCOMMON_MESSAGE_H_
+
+#include <stdint.h>
+
+/**
+ * @name Indexes of the log messages
+ *
+ * @added 1.6.0
+ *
+ */
+enum xkb_message_index {
+    {% for entry in entries %}
+    /** {{ entry.description }} */
+    {{ entry|message_index }},
+    {% endfor %}
+    XKB_LOG_MESSAGE_MAX_INDEX
+};
+
+typedef uint32_t xkb_message_code_t;
```

Comment:
Why do we need to distinguish between message code and message index?

Reply:
@bluetech does this approach seems right to you? I would like to have a good design for the message framework before starting to modify every log entry.

Reply:
I think it would be better to eliminate the "message index" concept, and use just the codes, in order to simplify things. Because the codes are not consecutive, it means that whenever they're looked up, will need to do the binary search on the array. But since it is absolutely not performance critical, and the array is quite small (even if it had 1000 entries it would still be small for that matter), I don't think that's a problem.

However, if you prefer to keep the indexes, that's OK by me.

Reply:
Removed the index concept. In fact we do not need to do the lookup.